pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -608,15 +597,22 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 	case hivev1.CreateOnlySyncSetApplyBehavior:
 		applyFn = h.Create
 	}
-
+	// determine if we can gather info for the resource and apply it
 	var applyErr error
-	for i, resource := range resources {
+	var discoveryErr error
+	info := &hiveresource.Info{}
+	for _, resource := range resources {
+		u := &unstructured.Unstructured{}
+		if err := yaml.Unmarshal(resource.Raw, u); err != nil {
```

Comment:
I think we had a misundestanding about where we landed on this. Since we need to do the `h.Info` call anyway in order to get the Resource of the object, there is no point in doing the yaml unmarshaling.

Reply:
yaml unmarshalling was picked up from Devan's PR. I think we are good to use h.Info object here, @dgoodwin Is there any specific reason you want to have unmarshalling? Can we go ahead and remove it?

Reply:
We can probably forego the explicit yaml unmarshaling and instead return in the `Info` the `Unstructured` object built when the `Info` object is created.

Reply:
Sounds good to me.

Reply:
@staebler I do not see any `Unstructured` object getting built when the `Info` object gets created in Info.go. Am I missing something here?

Reply:
I was thinking something along the lines of the following.
https://github.com/staebler/hive/commit/600fd9a568191d6fd8643496108f60857c7aaffb

Reply:
Since we have decided to move UnknownObjectSync condition to individual resource now, doesn't it make sense to have resourceSyncStatus object populated with minimum info before we set any condition on it. I think unmarshaling before calling info will help in here. Does it affect performance wise?

Reply:
I don't think that is necessary. Resource *i* in the status will correspond to resource *i* in the spec. The API version, kind, name, and namespace in the resource in the status are there for correlating applied resources to resources in the spec on subsequent reconciles in case the resources in the spec have changed position in the list. In this case, that is not an issue since the resource will not be applied.